components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java

```diff
@@ -0,0 +1,263 @@
+        } catch (RegistryException | UnsupportedEncodingException e) {
+            String error = "Error creating a registry resource from contents of %s email template type in %s locale.";
+            throw new I18nEmailMgtServerException(String.format(error, templateDisplayName, locale), e);
+        }
+
+        return templateResource;
+    }
+
+    /**
+     * @param templateResource
+     * @return
+     * @throws I18nEmailMgtException
+     */
+    public static EmailTemplate getEmailTemplate(Resource templateResource) throws I18nEmailMgtException {
+        EmailTemplate templateDTO = new EmailTemplate();
+        try {
+            // process email template meta-data properties
+            String templateDisplayName = templateResource.getProperty(I18nMgtConstants.TEMPLATE_TYPE_DISPLAY_NAME);
+            String templateType = templateResource.getProperty(I18nMgtConstants.TEMPLATE_TYPE);
+            String contentType = templateResource.getProperty(I18nMgtConstants.TEMPLATE_CONTENT_TYPE);
+            String locale = templateResource.getProperty(I18nMgtConstants.TEMPLATE_LOCALE);
+
+            templateDTO.setTemplateDisplayName(templateDisplayName);
+            templateDTO.setTemplateType(templateType);
+            templateDTO.setEmailContentType(contentType);
+            templateDTO.setLocale(locale);
+
+            // process email template content
+            Object content = templateResource.getContent();
+            if (content != null) {
+                byte templateContentArray[] = (byte[]) content;
+                String templateContent = new String(templateContentArray, Charset.forName("UTF-8"));
+                String[] templateContentElements = StringUtils.split(templateContent, "|");
+
+                // TODO should find a better way to maintain sections of email template.
+                if (templateContentElements.length > 3) {
+                    String errorMsg = "Template %s:%s contains '|' character which is invalid.";
+                    throw new I18nMgtEmailConfigException(String.format(errorMsg, templateDisplayName, locale));
+                } else if (templateContentElements.length < 3) {
+                    String errorMsg = "Template %s:%s contains empty body content.";
+                    throw new I18nMgtEmailConfigException(String.format(errorMsg, templateDisplayName, locale));
+                }
+
+                templateDTO.setSubject(templateContentElements[0]);
+                templateDTO.setBody(templateContentElements[1]);
+                templateDTO.setFooter(templateContentElements[2]);
+            } else {
+                String error = String.format("Unable to find any content in %s:%s email template.",
+                        templateDisplayName, locale);
+                log.error(error);
+            }
+        } catch (RegistryException e) {
+            String error = "Error when creating template object from registry resource";
```

Comment:
Wrong error message logging. Error log should be related to a error associated with retrieving a resource.